src/react-imgix.js

```diff
@@ -63,43 +39,89 @@ export default class ReactImgix extends Component {
     disableLibraryParam: PropTypes.bool
   };
   static defaultProps = {
-    aggressiveLoad: false,
     auto: ["format"],
     entropy: false,
     faces: true,
     fit: "crop",
-    fluid: true,
-    generateSrcSet: true,
+    disableSrcSet: false,
     onMounted: () => {},
-    precision: 100,
     type: "img"
   };
-  state = {
-    width: null,
-    height: null,
-    mounted: false
-  };
 
-  forceLayout = () => {
+  componentDidMount = () => {
     const node = ReactDOM.findDOMNode(this);
-    this.setState({
-      width: node.scrollWidth,
-      height: node.scrollHeight,
-      mounted: true
-    });
     this.props.onMounted(node);
   };
 
-  componentDidMount = () => {
-    this.forceLayout();
-  };
+  buildSrcs = () => {
+    const props = this.props;
+    const {
+      width,
+      height,
+      entropy,
+      faces,
+      auto,
+      customParams,
+      disableLibraryParam,
+      disableSrcSet,
+      type
+    } = props;
+
+    let crop = false;
+    if (faces) crop = "faces";
+    if (entropy) crop = "entropy";
+    if (props.crop) crop = props.crop;
+
+    let fit = false;
+    if (entropy) fit = "crop";
+    if (props.fit) fit = props.fit;
 
-  _findSizeForDimension = dim =>
-    findSizeForDimension(dim, this.props, this.state);
+    const fixedSize = width != null || height != null;
+
+    const srcOptions = {
+      auto,
+      ...customParams,
+      crop,
+      fit,
+      ...(disableLibraryParam ? {} : { ixlib: `react-${PACKAGE_VERSION}` }),
+      ...(fixedSize && height ? { height } : {}),
+      ...(fixedSize && width ? { width } : {})
+    };
+
+    const src = constructUrl(this.props.src, srcOptions);
+
+    let srcSet;
+
+    if (disableSrcSet) {
+      srcSet = src;
+    } else {
+      if (fixedSize || type === "source") {
+        const dpr2 = constructUrl(this.props.src, { ...srcOptions, dpr: 2 });
+        const dpr3 = constructUrl(this.props.src, { ...srcOptions, dpr: 3 });
+        srcSet = `${dpr2} 2x, ${dpr3} 3x`;
+      } else {
+        const buildSrcSetPair = targetWidth => {
+          const url = constructUrl(this.props.src, {
+            ...srcOptions,
+            width: targetWidth
+          });
+          return `${url} ${targetWidth}w`;
+        };
+        const addFallbackSrc = srcSet => srcSet.concat(`, ${src}`);
+        const _srcSet = targetWidths.map(buildSrcSetPair).join(", ");
+        const _srcSetWithFallback = addFallbackSrc(_srcSet);
+        srcSet = _srcSetWithFallback;
```

Comment:
This seems like a really convoluted way of doing this. How about just doing this?

```js
srcSet = targetWidths.map(buildSrcSetPair).concat([src]).join(", ");
```

Reply:
My aim with `_srcSetWithFallback` was to make clear what the added `src` was for. Would this work?

```js
        const addFallbackSrc = srcSet => srcSet.concat(src);
        const _srcSet = targetWidths.map(buildSrcSetPair);
        const _srcSetWithFallback = addFallbackSrc(_srcSet);
        srcSet = _srcSetWithFallback.join(', ');
```

or even

```js
        const addFallbackSrc = srcSet => srcSet.concat(src);
        srcSet = addFallbackSrc(targetWidths.map(buildSrcSetPair)).join(', ');
```